clang/test/Sema/attr-lifetime-suppress.cpp

```diff
@@ -0,0 +1,42 @@
+// RUN: %clang_cc1 -fcxx-exceptions -fsyntax-only -verify -Wlifetime -Wno-dangling %s
+
+namespace {
+template <class T>
+class [[gsl::Owner(T *)]] unique_ptr {
+public:
+  unique_ptr(T * t) : t(t) {}
+
+  T *release() {
+    T *tmp = t;
+    t = nullptr;
+    return tmp;
+  }
+
+  T *t = nullptr;
+};
+
+class PrivateKey {
+  // A random class that models a private key.
+};
+} // namespace
+
+namespace nosuppressOnFunction {
+PrivateKey *getPrivateKey() {
+  return unique_ptr<PrivateKey>(new PrivateKey()).release(); // expected-warning {{returning a dangling pointer}}
+  // expected-note@-1 {{temporary was destroyed at the end of the full expression}}
```

Comment:
I see that you closely model the issue in your test. I'm concerned that we might change the the behavior of the `unique_ptr::release()` handling and that the `suppress` attribute test shouldn't really depend on this.
Maybe 
```
int* f() {
  int* p;
  return p;
}
```
is a more concise way to produce a "returning a dangling pointer" issues and testing the [[gsl::suppress("lifetime")]] behavior? Then you also don't need the unique_ptr and PrivateKey scaffolding.

Reply:
Reworked the test case